packages/fast-form-generator-react/README.md

```diff
@@ -0,0 +1,181 @@
+
+# ReactJS FAST form generator
+A self generating UI based on JSON Schemas.
+
+It can be utilized to dynamically generate a form for changing data values on a React component.
+
+## Installing
+- Run `npm i`
+
+## Getting started
+- Run `npm start`
+
+## Using the form generator
+### Basic usage
+The required properties are the data, schema and onChange function. The data should be tied to your state, this will change.
+
+```jsx
+import Form from "@microsoft/fast-form-generator-react";
+
+<Form
+    data={this.state.currentComponentData}
+    schema={currentComponentSchema}
+    onChange={handleChange}
+/>
+```
+
+### onChange
+The onChange is used as a callback, it should take one argument that is the data taht should be updated when any data has been changed from within the generated form.
+
+example onChange:
+```jsx
+/**
+ * The app on change event
+ */
+onChange = (data) => {
+    this.setState({
+        currentComponentData: data
+    });
+}
+```
+
+## Writing JSON Schemas
+The schema form generator can interpret most [JSON schemas](http://json-schema.org/), however there are some things to note when writing JSON schemas that make for a better UI.
+
+### title
+Using a title will add a label to the left or top of the corresponding form element. All properties are required to have a title.
+
+Example:
+```
+{
+    "$schema": "http://json-schema.org/schema#",
+    "id": "my-component",
+    "title": "My component",
+    "type": "object",
+    "properties": {
+        "text": {
+            "title": "Text",
+            "type": "string",
+            "example": "Hello world"
+        },
+        "weight": {
+            "title": "Weight",
+            "type": "string",
+            "enum": [
+                "heavy"
+            ]
+        }
+    },
+    "required": [
+        "text"
+    ]
+}
+```
+
+### examples & default
+If an example or a default value is given, that is what will be used instead of 'example text' in case of a string, or a randomly generated number instead of a number. It is generally better to add this extra information in case the schema form generator needs to create a new set of data.
```

Comment:
Would replace the first sentence with: "Providing an example or default value will replace the placeholder 'example text' or randomly generated number."

Reply:
fixed